src/include/network/lan/address.rb

```diff
@@ -1631,5 +1611,36 @@ def repeated_physical_port_ids(slaves)
 
       physical_port_ids
     end
+
+    # Performs hostname update
+    #
+    # @param [String] ip address
+    # @param [String] new hostname
+    def update_hostname(ipaddr, hostname)
+      ip_changed = LanItems.ipaddr != ipaddr
+      hostname_initial = hostname_initial(LanItems.ipaddr)
+      hostname_changed = hostname_initial != hostname
+
+      return if !(ip_changed || hostname_changed || hostname.empty?)
+
+      log.info("Dropping record for #{LanItems.ipaddr} from /etc/hosts")
+
+      Host.remove_ip(LanItems.ipaddr)
+      Host.Update(hostname_initial, hostname, ipaddr) if !hostname.empty?
+
+      nil
```

Comment:
new method and you do not use return statement, so this is not needed ( legacy ycp conversion )